tracing-mock/tests/readme.rs

```diff
@@ -0,0 +1,78 @@
+use tracing::collect::with_default;
```

Comment:
i think that rather than having these examples duplicated in a separate tests file, we might want to consider putting 
```rust
#![doc = include_str!("../README.md)]
```
in the `lib.rs` file for `tracing-mock`. That way, the examples in the readme would be run as doctests, and we wouldn't have to deal with duplication between the README and the lib.rs docs.

We should probably be doing this for the other `tracing` crates, instead of duplicating the same documentation between the readme and `lib.rs`, but I haven't had the time to switch everything over yet...

Reply:
+1

Reply:
Ohhh, that's a nice approach. I was trying to work out how to do something like that, but I didn't think of that. I'll give it a try and push an update.

Reply:
Oh, one thing I just realised: The examples in the README are for the `v0.1.x` branch, following the example from the README for `tracing` itself. So those examples won't actually pass (or even compile) on `master`.

They can be substituted for examples for `master`, but that might be (more) confusing for people arriving at the repo for the first time.

Reply:
As discussed on Discord, the examples in the README are now for the `master` branch codeline. The same will be done for the other crates' README files.